packages/block-library/src/post-title/index.php

```diff
@@ -19,7 +19,19 @@ function render_block_core_post_title( $attributes, $content, $block ) {
 		return '';
 	}
 
-	return '<h1>' . get_the_title( $block->context['postId'] ) . '</h1>';
+	$tag_name = 'h2';
+	$align_class_name  = empty( $attributes['align'] ) ? '' : ' ' . "has-text-align-{$attributes['align']}";
+
+	if ( isset( $attributes['level'] ) ) {
+		$tag_name = 0 === $attributes['level'] ? 'p' : 'h' . $attributes['level'];
+	}
+
+	return sprintf(
+		'<%1$s class="%2$s">%3$s</%1$s>',
+		$tag_name,
+		'wp-block-post-title' . esc_attr( $align_class_name ),
+		get_the_title( $block->context['postId'] )
+	);
```

Comment:
As an aside or a potential follow-up:

The "align" support flag automatically handles  the addition of the className for static blocks (save functions), but  it doesn't do so on the server. Now that we have server-side registration for most blocks, we could write some hooks to do this automatically. It involves some Regex work but I think it's doable. cc @gziolo 

(same for other hooks like the custom className...)

Reply:
@aduth had the logic for what you described in one of his PRs :) The challenge is that we would make sure that the client and server always stay in sync. It's not only "align" and "className", but also colors, fonts, etc.

Reply:
Actually, I have the logic here too https://github.com/WordPress/gutenberg/pull/21420

Reply:
+1 to support things, I'm looking to add color, font and line height support to the Site Title block and it needs server side support. :+1:

Reply:
> @aduth had the logic for what you described in one of his PRs :)

It was in #18414 (specifically https://github.com/WordPress/gutenberg/pull/18414/commits/c7cded60c73e8e5c784743bc3474a0914c55928e).

Reply:
@youknowriad I've opened the Site Title block PR that would need support on the server side for classnames: https://github.com/WordPress/gutenberg/pull/23007

What needs to be done to get that support in master? I'm very happy to help with it.

Reply:
@apeatling I think right now the server side block registration already knows about the "supports" key (@gziolo confirm?), which means what's remaining is to write a `render_block` filter like that https://github.com/WordPress/gutenberg/pull/21420/files#diff-4339a9e86722fd029c231f3fb65f0ad2R392-R420 to apply the styles and classNames.

Reply:
Started working on this here https://github.com/WordPress/gutenberg/pull/23007/files#diff-1334f2cb73e38f2f89d2aaf763f07ed9R307